rfs/src/store/dir.rs

```diff
@@ -49,9 +57,13 @@ impl Store for DirStore {
     }
 
     async fn set(&self, key: &[u8], blob: &[u8]) -> Result<()> {
-        let path = self.root.join(hex::encode(key));
+        let file_name = hex::encode(key);
+        let dir_path = self.root.join(&file_name[0..2]);
+
+        fs::create_dir_all(dir_path.clone()).await?;
```

Comment:
```suggestion
        fs::create_dir_all(&dir_path).await?;
```

I see u use clone in places where a & can be passed and is more efficient to do so.
always read docs of functions before u use them, Some of them takes `AsRef<T>` which means this can either take T, or &T. Will save you a clone